SingularityUI/app/components/requestForm/RequestForm.jsx

```diff
@@ -0,0 +1,564 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import FormField from '../common/formItems/FormField';
+import DropDown from '../common/formItems/DropDown';
+import CheckBox from '../common/formItems/CheckBox';
+import { modifyField, clearForm } from '../../actions/form';
+import {SaveAction} from '../../actions/api/request';
+import OverlayTrigger from 'react-bootstrap/lib/OverlayTrigger';
+import ToolTip from 'react-bootstrap/lib/Tooltip';
+import Utils from '../../utils';
+import classNames from 'classnames';
+
+const QUARTZ_SCHEDULE = 'quartzSchedule';
+const CRON_SCHEDULE = 'cronSchedule';
+
+const FORM_ID = 'requestForm';
+
+const REQUEST_TYPES = ['SERVICE', 'WORKER', 'SCHEDULED', 'ON_DEMAND', 'RUN_ONCE'];
+
+const FIELDS_BY_REQUEST_TYPE = {
+  ALL: [
+    'id',
+    'owners',
+    'requestType',
+    'slavePlacement'
+  ],
+  SERVICE: [
+    'instances',
+    'rackSensitive',
+    'hideEvenNumberAcrossRacksHint',
+    'loadBalanced',
+    'rackAffinity'
+  ],
+  WORKER: [
+    'instances',
+    'rackSensitive',
+    'hideEvenNumberAcrossRacksHint',
+    'waitAtLeastMillisAfterTaskFinishesForReschedule',
+    'rackAffinity'
+  ],
+  SCHEDULED: [
+    QUARTZ_SCHEDULE,
+    CRON_SCHEDULE,
+    'scheduleType',
+    'numRetriesOnFailure',
+    'killOldNonLongRunningTasksAfterMillis',
+    'scheduledExpectedRuntimeMillis'
+  ],
+  ON_DEMAND: [ 'killOldNonLongRunningTasksAfterMillis' ],
+  RUN_ONCE: [ 'killOldNonLongRunningTasksAfterMillis' ]
+};
+
+class RequestForm extends React.Component {
+
+  componentDidMount() {
+    this.props.clearForm(FORM_ID);
+  }
+
+  componentWillReceiveProps(newProps) {
```

Comment:
This isn't an idiomatic way to update the URL. Leverage the fact that dispatching the save action returns the original XHR promise:
```javascript
save(requestId) {
    dispatch(SaveAction.trigger(requestBody)).then(() => { Backbone.history.navigate(`blah`); });
}
```